### PR TITLE
#67491 implement integration tests to validate config refresh endpoint

### PR DESCRIPTION
**Pull request by tabman83**
Implement integration tests to validate the refresh configuration endpoint.

### CODE REVIEW

File: src/CaptainHook.Api.Tests/config/ApiClientFixture.cs [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
+        private static Uri CaptainHookTestUri = new Uri("https://localhost:24010");
+
         public ICaptainHookClient GetApiUnauthenticatedClient()
         {
-            var url = new Uri("https://localhost:24010");
+            return new CaptainHookClient(CaptainHookTestUri, AnonymousCredential.Instance);
+        }
 
-            return new CaptainHookClient(url, AnonymousCredential.Instance);
+        //private TokenCredentials CreateCredentials()
```
Comment (dawidkeshop) [inline src/CaptainHook.Api.Tests/config/ApiClientFixture.cs:17]: I think it's better to remove commented code block.

Comment (tabman83) [reply src/CaptainHook.Api.Tests/config/ApiClientFixture.cs:17]: Thanks just forgot.

File: src/CaptainHook.Api.Client/scripts/GenerateClient.ps1 [hunk 2]
```diff
@@ -76,7 +76,7 @@ function Install-Tooling
 	param([string] $ToolsPath)
 
 	Write-Output "Installing or updating Swashbuckle CLI"
-	dotnet tool install swashbuckle.aspnetcore.cli --version 5.4.1 --tool-path $ToolsPath
+	dotnet tool update swashbuckle.aspnetcore.cli --version 5.4.1 --tool-path $ToolsPath
```
Comment (mszymeshop) [inline src/CaptainHook.Api.Client/scripts/GenerateClient.ps1:79]: :+1:

File: src/CaptainHook.Api.Tests/appsettings.CI.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "StsApiEndpoint": "https://security-sts-api.ci.eshopworld.net",
```
Comment (mszymeshop) [inline src/CaptainHook.Api.Tests/appsettings.CI.json:2]: I have doubts if we should be exposing these publically. My preference would be to use KeyVault to store these instead. @chomick @dguerin @tabman83 what do you think?

Comment (VaclavK) [reply src/CaptainHook.Api.Tests/appsettings.CI.json:2]: already exposed :D

File: src/CaptainHook.Api.Tests/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "RefreshingTokenProviderOptions": {
+    "ClientId": "tooling.eda.api.client",
```
Comment (mszymeshop) [inline src/CaptainHook.Api.Tests/appsettings.json:3]: Should we change the client id name? Kevin has mentioned this is too similar to the tooling.eda.client but I am not sure if there is a better name. What do you think?

Comment (VaclavK) [reply src/CaptainHook.Api.Tests/appsettings.json:3]: I personally have no problem decyphering the difference and actually the similar nature makes sense in terms of tokenisation.... my 2 c

File: src/CaptainHook.Api.Tests/appsettings.CI.json (hunk 3 above)
Comment (mszymeshop) [reply src/CaptainHook.Api.Tests/appsettings.CI.json:2]: @VaclavK haha, well yeah :) Still, what do you think here?

Comment (VaclavK) [reply src/CaptainHook.Api.Tests/appsettings.CI.json:2]: hiding the url like that does not IMHO add real security - it is essentially security by obscurity..

either we have a hole (and in that sense hiding the URL (already known I am sure even from elsewhere) is not going to help or we do not and we are fine...

Comment (tabman83) [reply src/CaptainHook.Api.Tests/appsettings.CI.json:2]: I don't think a URL is a leak as it's an already publicly available resource, but I am happy to take your guidance here.

File: src/CaptainHook.Api.Tests/appsettings.json (hunk 4 above)
Comment (tabman83) [reply src/CaptainHook.Api.Tests/appsettings.json:3]: I don't have a strong opinion here so I am happy to take your guidance here as well.

File: src/CaptainHook.Api.Tests/appsettings.CI.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (chomick) [reply src/CaptainHook.Api.Tests/appsettings.CI.json:2]: needed for token content

File: src/CaptainHook.Api.Tests/appsettings.PROD.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "StsApiEndpoint": "https://security-sts-api.production.eshopworld.com",
```
[the remark this answers is not in the document]
Comment (tabman83) [reply src/CaptainHook.Api.Tests/appsettings.PROD.json]: I have removed settings file for both PROD and SAND environments.

File: src/CaptainHook.DirectorService/DirectorService.cs [hunk 6]
```diff
@@ -146,6 +146,11 @@ protected override async Task RunAsync(CancellationToken cancellationToken)
 
         public Task ReloadConfigurationForEventAsync(string eventName)
         {
+            if(string.IsNullOrWhiteSpace(eventName))
```
[the remark this answers is not in the document]
Comment (tabman83) [reply src/CaptainHook.DirectorService/DirectorService.cs]: I have removed the event name validation for now.

File: src/CaptainHook.Api.Tests/config/ApiClientFixture.cs [hunk 7]
```diff
@@ -1,16 +1,23 @@
 ﻿using System;
 using CaptainHook.Api.Client;
+using EShopworld.Security.Services.Testing.Token;
 using Microsoft.Rest;
 
 namespace CaptainHook.Api.Tests.Config
 {
     public class ApiClientFixture
     {
+        private static Uri CaptainHookTestUri = new Uri("https://localhost:24010");
```
Comment (chomick) [inline src/CaptainHook.Api.Tests/config/ApiClientFixture.cs:10]: Should that not be configured per env?

File: src/CaptainHook.Api/Controllers/RefreshConfigController.cs [hunk 8]
```diff
@@ -31,6 +35,10 @@ public async Task<IActionResult> RefreshConfigForEvent([FromBody]RefreshConfigRe
 
                 return Ok();
             }
+            catch(Exception exception) when (exception.InnerException is ArgumentNullException)
```
[the remark this answers is not in the document]
Comment (tabman83) [reply src/CaptainHook.Api/Controllers/RefreshConfigController.cs]: This has been removed following our conversation.

File: src/CaptainHook.Api/Controllers/RefreshConfigController.cs [hunk 9]
```diff
@@ -31,6 +35,10 @@ public async Task<IActionResult> RefreshConfigForEvent([FromBody]RefreshConfigRe
...
+            {
+                return NotFound();
+            }
             catch
```
[the remark this answers is not in the document]
Comment (tabman83) [reply src/CaptainHook.Api/Controllers/RefreshConfigController.cs]: The generic exception is now being logged.

File: src/CaptainHook.Api.Tests/config/ApiClientFixture.cs (hunk 7 above)
Comment (tabman83) [reply src/CaptainHook.Api.Tests/config/ApiClientFixture.cs:10]: Test is marked as IsDev now as suggested by @mszymeshop

File: src/CaptainHook.Api/Controllers/RefreshConfigController.cs [hunk 10]
```diff
@@ -31,8 +47,9 @@ public async Task<IActionResult> RefreshConfigForEvent([FromBody]RefreshConfigRe
 
                 return Ok();
             }
-            catch
+            catch(Exception exception)
```
Comment (chomick) [inline src/CaptainHook.Api/Controllers/RefreshConfigController.cs:50]: Normally would not agree that this is the correct behaviour - if the exception happens that we do not understand, we should fail with 5xx or simply let exception to bubble up - no reason to catch this for the sake of logging and returning 400.
But to move on with this PR and as this is temp endpoint, I think we are ok. Just a note for future work.

File: src/CaptainHook.Api.Tests/config/ApiClientFixture.cs (hunk 7 above)
Comment (chomick) [reply src/CaptainHook.Api.Tests/config/ApiClientFixture.cs:10]: ok, so we do not expect htem to run on CI? I think this is not correct, we should to ensure the deployment is healthy?
Again, happy to close it for now but we need a significant re-do of the testing in CH - it is very messy right now - we should follow the same standards as every other team/project follows, in relation to naming conventions, folder structures and test attributes.

File: src/CaptainHook.Api/Controllers/RefreshConfigController.cs (hunk 10 above)
Comment (tabman83) [reply src/CaptainHook.Api/Controllers/RefreshConfigController.cs:50]: Noted, thanks.

**Pull request merged**